paimon-core/src/main/java/org/apache/paimon/index/PartitionIndex.java

```diff
@@ -137,4 +149,47 @@ public static PartitionIndex loadIndex(
         }
         return new PartitionIndex(mapBuilder.build(), buckets, targetBucketRowNumber);
     }
+
+    public static int cacheBucketAndGet(Int2ShortHashMap hash2Bucket, int hash, int bucket) {
```

Comment:
Don't need to extract this method, it is too simple.

Reply:
done :)